external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala

```diff
@@ -202,4 +202,46 @@ class MsSqlServerIntegrationSuite extends DockerJDBCIntegrationSuite {
     df2.write.jdbc(jdbcUrl, "datescopy", new Properties)
     df3.write.jdbc(jdbcUrl, "stringscopy", new Properties)
   }
+
+  test("SPARK-29644: Write tables with ShortType") {
+    import testImplicits._
+    val df = Seq(-32768.toShort, 0.toShort, 1.toShort, 38.toShort, 32768.toShort).toDF("a")
+    val tablename = "shorttable"
+    df.write
+      .format("jdbc")
+      .mode("overwrite")
+      .option("url", jdbcUrl)
+      .option("dbtable", tablename)
+      .save()
+    val df2 = spark.read
+      .format("jdbc")
+      .option("url", jdbcUrl)
+      .option("dbtable", tablename)
+      .load()
+    assert(df.count == df2.count)
+    val rows = df2.collect()
+    val colType = rows(0).toSeq.map(x => x.getClass.toString)
+    assert(colType(0) == "class java.lang.Short")
+  }
+
+  test("SPARK-29644: Write tables with ByteType") {
```

Comment:
I think its ok for `JDBCWriteSuite` to only have the two write tests below for byte and short. So, can you remove the write tests from the integration tests?

Reply:
I think both test have their important purpose. JDBCWriteSuite unit test will catch any spark regression in CI/CD and integration test are great to ensure that functionality works end to end. Consider that ByteType should translate to TINYINT in SQLServer, but in PostGres  both ByteType maps to SMALLINT. This real distinction is factually verified in Integration test. So i think having integration test is very important and thus we should not remove these.

Reply:
But, very verbose. So, can we share this test by making a base abstract test class?

Reply:
Agree. Will take a shot at it today,  else will remove these test cases and attempt add these cleanly with fix later.

Reply:
Several test cases in the integration suite are quite similar, the framework does not have a notion of a common base class for same test case. This fix will require some refactoring of the framework. Something i can attempt later, not in scope of this PR. 

I would remove the repeated test in all DB types and retain it for SQL server only.

Reply:
Fixed. These repeated test cases are removed. Only retained in MsSQLIntegrationTest. Will refactor test suite later to have common test cases in it.

Reply:
Ur, sorry for my late response. Yea, that follow-up looks ok to me.